### PR TITLE
feat: floor price change percentage

### PR DESCRIPTION
**Pull request by alfonsobries**
## Summary

Closes https://app.clickup.com/t/86dqrnkyz

- adds price history table and store new entries every time a new floor price is fetched
- Calculates the floor price change by using the average price from today vs the average day from yesterday
- updates the frontend to reflect those percentages and shows `N/A` if we dont have the data
- im not considering any kind of pruning for now, we can handle that if table starts to grow up too much

## Checklist

-   [ ] I checked my UI changes against the design and there are no notable differences
-   [ ] I checked my UI changes for any responsiveness issues
-   [ ] I checked my (code) changes for obvious issues, debug statements and commented code
-   [ ] I opened a corresponding card on Clickup for any remaining TODOs in my code
-   [ ] I added a short description on how to test this PR _(if necessary)_
-   [ ] I added a storybook entry for the component that was added _(if necessary)_
-   [ ] Documentation _(if necessary)_
-   [ ] Tests _(if necessary)_
-   [ ] Ready to be merged

### CODE REVIEW

File: database/migrations/2023_12_11_202944_create_floor_price_history_table.php [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+return new class extends Migration
+{
+    public function up(): void
+    {
+        Schema::create('floor_price_history', function (Blueprint $table) {
+            $table->id();
+            $table->foreignIdFor(Collection::class)->constrained()->cascadeOnDelete();
```
Comment (oXtxNt9U) [inline database/migrations/2023_12_11_202944_create_floor_price_history_table.php:17]: We should make sure this column has an index (and also `retrieved_at`)

File: database/migrations/2023_12_11_202944_create_floor_price_history_table.php [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+use App\Models\Token;
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+return new class extends Migration
+{
+    public function up(): void
+    {
+        Schema::create('floor_price_history', function (Blueprint $table) {
```
Comment (oXtxNt9U) [inline database/migrations/2023_12_11_202944_create_floor_price_history_table.php:15]: Does it make sense for `floor_price` and `token_id` to be nullable? Couldn't we instead just not store anything if we don't have either of them?

**Review by oXtxNt9U — CHANGES_REQUESTED** [review]

File: database/migrations/2023_12_11_202944_create_floor_price_history_table.php (hunk 2 above)
Comment (alfonsobries) [reply database/migrations/2023_12_11_202944_create_floor_price_history_table.php:15]: @oXtxNt9U yes, good catch

**Review by crnkovic — COMMENTED** [review]
I like this approach. One question: does the `retrieved_at` column need to be `nullable`? Because essentially, `retrieved_at` is current time, right? I know we had some logic for null values in the job class, but can we ditch that? Can `retrieved_at` be anything else other than `NOW()` when storing? I don't understand why we remove time if token is missing (nothing with your PR, just thinking out loud as we do that prior to this PR as well).

Also, if `retrieved_at` is `null`, have you tested that it calculates values properly? Since you compare with `CURRENT_DATE`, etc.

**Comment by alfonsobries** [comment]
@crnkovic retrieved_at shouldnt be null, just fixed it

**Review by crnkovic — APPROVED** [review]

**Review by oXtxNt9U — APPROVED** [review]

**Comment by crnkovic** [comment]
Added the task for pruning old changes: https://app.clickup.com/t/86dqum8fj

**Pull request merged**